17-switch-mass-hypo.md

```diff
@@ -10,5 +10,81 @@ minutes: 10
+
+Next we have to handle the input and output of this algorithm. This is accomplished using [particle selections](https://twiki.cern.ch/twiki/bin/view/LHCb/ParticleSelection). The input to our substitution algorithm will be the candidates produced by the stripping. In order to make them look like a selection we can use the DataOnDemand service:
+```python
+# Stream and stripping line we want to use
+stream = 'AllStreams'
+line = 'D2hhCompleteEventPromptDst2D2RSLine'
+tesLoc = '/Event/{0}/Phys/{1}/Particles'.format(stream, line)
+
+# get the selection(s) created by the stripping
+from PhysSelPython.Wrappers import Selection
+from PhysSelPython.Wrappers import SelectionSequence
+from PhysSelPython.Wrappers import DataOnDemand
+
+strippingSels = [DataOnDemand(Location=tesLoc)] 
+```
+
+The output of the algorithm has to be packaged into a new selection:
+
+```python
+# create a selection using the substitution algorithm
+selSub = Selection("Dst2D0pi_D02pipi_Sel", Algorithm=Subs, RequiredSelections=strippingSels)
+```
+
+Note how the input stripping selection is daisy-chained to the output selection through the RequiredSelections (it has to be a list of selections) argument. The new selection is added into a SelectionSequnce for further use by DaVinci:
+
+```python
+selSeq = SelectionSequence('SelSeq', TopSelection=selSub)
+```
+
+We are now ready to produce an ntuple on our newly created selection. As usual we configure a DecayTreeTuple, which now is looking for the decay tree which contains the redefined D0 decay:
+```python
+# Create an ntuple to capture D*+ decays from the new selection
+dtt = DecayTreeTuple('TupleDstToD0pi_D0Topipi')
+dtt.Inputs = [selSeq.outputLocation()]
+dtt.Decay = '[D*(2010)+ -> ^(D0 -> ^pi- ^pi+) ^pi+]CC'  #note the redefined decay of the D0 
+```
+
+The input to the DecayTreeTuple is taken from the SelectionSequence we just created. 
+
+> ## Why use DaVinci selections? {.callout}
+> Selections and SelectionSequences are an elegant way to organize the required algorithms that perform the job of selecting data.
+> The [particle selection toolkit](https://twiki.cern.ch/twiki/bin/view/LHCb/ParticleSelection) uses python tricks to make the management of even complicated sequences of selections rather straight forward. In particular the toolkit ensures that all necessary algorithms are run in the correct order to produce the desired selections. It is therfore mainly a tool to manage dependencies. For technical details have a look at the [TWiki page](https://twiki.cern.ch/twiki/bin/view/LHCb/ParticleSelection).
+
+Finally we add the SelectionSequence and the DecayTreeTuple to the DaVinci application. Since we are adding more than one algorithm we need a GaudiSequencer that takes care of calling everything in the right order:
```

Comment:
Yes, in principle underscores are the way to go, that's why I made my
comment.

On Thu, May 28, 2015 at 11:47 AM, Kevin Dungs <notifications@github.com>
wrote:

> In 17-switch-mass-hypo.md
> <https://github.com/lhcb/first-analysis-steps/pull/58#discussion_r31218084>
> :
>
> > +
> > +We are now ready to produce an ntuple on our newly created selection. As usual we configure a DecayTreeTuple, which now is looking for the decay tree which contains the redefined D0 decay:
> > +```python
> > +# Create an ntuple to capture D*+ decays from the new selection
> > +dtt = DecayTreeTuple('TupleDstToD0pi_D0Topipi')
> > +dtt.Inputs = [selSeq.outputLocation()]
> > +dtt.Decay = '[D*(2010)+ -> ^(D0 -> ^pi- ^pi+) ^pi+]CC'  #note the redefined decay of the D0
> > +```
> > +
> > +The input to the DecayTreeTuple is taken from the SelectionSequence we just created.
> > +
> > +> ## Why use DaVinci selections? {.callout}
> > +> Selections and SelectionSequences are an elegant way to organize the required algorithms that perform the job of selecting data.
> > +> The [particle selection toolkit](https://twiki.cern.ch/twiki/bin/view/LHCb/ParticleSelection) uses python tricks to make the management of even complicated sequences of selections rather straight forward. In particular the toolkit ensures that all necessary algorithms are run in the correct order to produce the desired selections. It is therfore mainly a tool to manage dependencies. For technical details have a look at the [TWiki page](https://twiki.cern.ch/twiki/bin/view/LHCb/ParticleSelection).
> > +
> > +Finally we add the SelectionSequence and the DecayTreeTuple to the DaVinci application. Since we are adding more than one algorithm we need a GaudiSequencer that takes care of calling everything in the right order:
>
> Now it might already be too late but there actually are official
> guidelines for these kinds of questions:
> https://www.python.org/dev/peps/pep-0008/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lhcb/first-analysis-steps/pull/58/files#r31218084>.
>



-- 
Dr. Albert Puig Navarro
Laboratoire de Physique des Hautes Energies
Ecole Polytechnique Fédérale de Lausanne (EPFL)
BSP 614.4 (Cubotron UNIL) CH-1015 Lausanne
EPFL Phone: 021 6939808
CERN Phone: 72518